app/konnect/runtime-manager/gateway-runtime-kubernetes.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Set up a Kong Gateway Runtime on Kubernetes
+no_version: true
+---
+Set up a Kubernetes [runtime](/konnect/overview/#konnect-key-concepts-and-terminology)
+through the
+[{{site.konnect_short_name}} Runtime Manager](/konnect/runtime-manager) and
+configure your {{site.base_gateway}} instance to accept configuration from
+{{site.konnect_short_name}}. The Runtime Manager keeps track of all runtimes
+associated with the {{site.konnect_short_name}} SaaS account.
+
+<div class="alert alert-ee blue">
+<b>Note:</b> Kong does not host runtimes. You must install and host your own
+runtime instances.
+</div>
+
+## Prerequisites
+
+* You have a {{site.konnect_product_name}} account. Contact your sales
+representative for access.
+* **Kubernetes cluster with load balancer:** {{site.konnect_short_name}} is
+compatible with all distributions of Kubernetes. You can use a Minikube, GKE,
+or OpenShift cluster.
+* **kubectl or oc access:** You should have kubectl or oc (if working with OpenShift)
+installed and configured to communicate to your Kubernetes cluster.
+* [Helm 3](https://helm.sh/docs/intro/install/) is installed.
+
+## Set up Helm
+
+On your runtime's system, create a namespace and pull down the `kong` Helm repo.
+
+1. Create a namespace:
+    ```sh
+    $ kubectl create namespace kong
+    ```
+
+2. Add the Kong charts repository:
+    ```bash
+    $ helm repo add kong https://charts.konghq.com
+    ```
+
+3. Update Helm:
+    ```bash
+    $ helm repo update
+    ```
+
+## Generate certificates
+
+{% include /md/konnect/runtime-certs.md %}
+
+## Configure the runtime
+
+Next, configure a {{site.base_gateway}} instance using the certificate, the
+private key, and the remaining configuration details on the
+**Configure Runtime** page.
+
+### Configure secrets
+
+Store the certificates and key you generated through the Runtime Manager in
+Kubernetes secrets.
+
+1. Create a `tls` secret using the `cluster.cert` and `cluster.key` files
+you saved earlier:
+
+    ```bash
+    $ kubectl create secret tls kong-cluster-cert -n kong \
+      --cert=/<path-to-file>/cluster.crt \
+      --key=/<path-to-file>/cluster.key
+    ```
+
+2. Create a generic secret for the `ca.crt` file:
+
+    ```bash
+    $ kubectl create secret generic kong-cluster-ca -n kong \
+      --from-file=ca.crt=/<path-to-file>/ca.crt
+    ```
+
+### Write configuration
+
+1. Back in {{site.konnect_short_name}}, copy the
+codeblock from the **Configuration Parameters** section.
+
+2. Create a `values.yaml` file. Remove the `KONG_` prefix from the parameters
+in the sample codeblock and add the following parameters to the file.
+
+    Make sure to replace the values in `cluster_cert`, `cluster_cert_key`,
+    and `cluster_ca_cert` with references to the secrets you created earlier:
+
```

Comment:
The `Write configuration` section could use some tweaks:
- this code block could be more specific there are multiple this could refer to
- remove the kong prefix is not clear enough as its also changes from upper to lower case, either way expecting a person following the docs to transform key casing introduces too much uncertainty. @skibum ideally, this should be in the advanced konnect tab ready to be copy pasted.
- replace the values isn't clear about what should be replaced

Reply:
> remove the kong prefix is not clear enough as its also changes from upper to lower case, either way expecting a person following the docs to transform key casing introduces too much uncertainty.

I see this as the biggest barrier to this deployment option. There is no level of documentation that is going to make this less confusing or error-prone, and it's a fairly simple change from a UI perspective (just remove the prefix and lower-case the variable names that are already shown). Even better would be a download option for a zip with a pre-filled `values.yaml` and all of the certs/keys ready to go.

Reply:
We are planning to update the UI to fix the casing and the prefix issue.

Reply:
Addressed bullet points 1 and 3 in doc - you caught the fact that the order of operations was just... not logical, so I reworked it. Point 2 I _sort of_ covered, but really it'll be fixed in the UI, and that will be way better.